### PR TITLE
🔨 Switch Search to use useOvermind

### PR DESCRIPTION
**Pull request by MichaelDeBoey**
Resubmission of #2852

Follow-up of #2780 & #2819 (both in `master`)

Things I did extra:
- Convert `elements.js` & `search.js` to TS
- Put all styles into the `elements.ts` file
- Export `Search` by a named export  instead of a `default export`
- Fix `Search`'s props by using `react-router-dom`'s `RouteComponentProps` generic instead of hardcoding it and using types from `history` (which is not our own dependency)

### CODE REVIEW

**Comment by lbogdan** [comment]
Build for latest commit https://github.com/codesandbox/codesandbox-client/pull/3048/commits/1795b3c4445259c4102beebe5c8c0d3d3275ba92 is at https://pr3048.build.csb.dev/s/new.

**Comment by SaraVieira** [comment]
Hey!

The code looks good but the styles are not applied in the pr build :(

https://pr3048.build.csb.dev/search

**Comment by SaraVieira** [comment]
update the ones in elements.ts are, the global ones are not

File: packages/app/src/app/pages/Search/search.ts [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+            }
+          }
+        }
+      }
+    }
+  `};
+`;
+
+export default createGlobalStyle`
+  body {
```
Comment (SaraVieira) [inline packages/app/src/app/pages/Search/search.ts]: Oh, its this, you need rempve this

**Comment by MichaelDeBoey** [comment]
@SaraVieira Looks like this one can be merged, no? 🤔

**Comment by SaraVieira** [comment]
Yap!

**Comment by MichaelDeBoey** [comment]
Always a pleasure! 🙂

**Pull request merged**